### PR TITLE
chore: Remove RN from lib-v1, add Util > Hub, fix TS

### PR DESCRIPTION
**Pull request by stocaaro**
#### Description of changes:
A few chores:
- Remove RN from dev preview menu
- Add Hub documentation
- Fix a linking issue in Troubleshooting

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib-v1/utilities/js/hub.mdx [hunk 1]
```diff
@@ -0,0 +1,94 @@
+Amplify has a local eventing system called Hub. It is a lightweight implementation of Publisher-Subscriber pattern, and is used to share data between modules and components in your app. Amplify uses Hub for different categories to communicate with one another when specific events occur, such as authentication events like a user sign-in or notification of a file download.
```
Comment (stocaaro) [inline src/fragments/lib-v1/utilities/js/hub.mdx:1]: This file is a direct copy from `lib` with minor edits (almost entirely removals) to make it V6 compatible.

File: src/fragments/lib-v1/utilities/js/hub.mdx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+* api
+* analytics
+* interactions
+* pubsub
+* storage
+* datastore
+
+### Listening for messages
+
+`listen(channel: string, callback: HubCallback): StopListenerCallback` is used to listen for messages that have been dispatched. You must provide either a named `channel` or a regular expression, along with a callback. In the case of a regular expression only dispatches which contain a `message` in their payload will be matched against your pattern. You can add multiple listeners to your application for different channels or patterns to listen for, or trap generic events and perform your own filtering.
```
Comment (stocaaro) [inline src/fragments/lib-v1/utilities/js/hub.mdx:24]: Pretty sure the type definition here is the only piece I changed.

File: src/fragments/lib-v1/utilities/js/hub.mdx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+A channel is a logical group name that you use to organize messages and listen on. These are strings and completely up to you as the developer to define for dispatching or listening. However, while you can dispatch to any channel, ***Amplify protects certain channels*** and will flag a warning as sending unexpected payloads could have undesirable side effects (such as impacting authentication flows). The protected channels are currently:
+
+* core
+* auth
+* api
+* analytics
+* interactions
+* pubsub
+* storage
+* datastore
```
Comment (stocaaro) [inline src/fragments/lib-v1/utilities/js/hub.mdx:20]: Should change shortened this list. Doing so now.

File: src/fragments/lib-v1/utilities/js/hub.mdx [hunk 4]
```diff
@@ -0,0 +1,90 @@
+Amplify has a local eventing system called Hub. It is a lightweight implementation of Publisher-Subscriber pattern, and is used to share data between modules and components in your app. Amplify uses Hub for different categories to communicate with one another when specific events occur, such as authentication events like a user sign-in or notification of a file download.
+
+## Installation
+```javascript
+import { Hub } from 'aws-amplify';
```
Comment (jimblanc) [inline src/fragments/lib-v1/utilities/js/hub.mdx:5]: Shouldn't this be exported from a sub-path? Did you find in on the root import?

```suggestion
import { Hub } from 'aws-amplify/utils';
```

File: src/fragments/lib-v1/utilities/js/hub.mdx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+* auth
+* analytics
+* storage
+
+### Listening for messages
+
+`listen(channel: string, callback: HubCallback): StopListenerCallback` is used to listen for messages that have been dispatched. You must provide either a named `channel` or a regular expression, along with a callback. In the case of a regular expression only dispatches which contain a `message` in their payload will be matched against your pattern. You can add multiple listeners to your application for different channels or patterns to listen for, or trap generic events and perform your own filtering.
+
+```javascript
+import { Hub } from 'aws-amplify';
```
Comment (jimblanc) [inline src/fragments/lib-v1/utilities/js/hub.mdx:23]: Ditto

File: src/fragments/lib-v1/utilities/js/hub.mdx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+`Hub.dispatch(channel: string, payload: HubPayload)` can be used to dispatch a `HubPayload` to a `channel`. The `channel` is a logical grouping for your organization while the `HubPayload` is a type defined as:
+```javascript
+export type HubPayload = {
+    event: string,
+    data?: any,
+    message?: string
+};
+```
+
+The `event` field is recommended to be a small string without spaces such as `signIn` or `hang_up` as it's useful for checking payload groupings. The `data` field is a freeform structure which many times is used for larger JSON objects or custom data types. Finally while `message` is optional, we encourage you to use it as it is required when using a `RegExp` filtering with `Hub.listen()`.
```
Comment (jimblanc) [inline src/fragments/lib-v1/utilities/js/hub.mdx:74]: Can remove ref to regex filtering

File: src/fragments/lib-v1/utilities/js/hub.mdx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+### Stop Listening
+
+Hub provides a way to stop listening for messages with calling the result of the `.listen` function. This may be useful if you no longer need to receive messages in your application flow, as well as to avoid any memory leaks on low powered devices when you are sending large amounts of data through Hub on multiple channels.
+
+To stop listening to a certain event, you need to wrap the listener function to a variable and call it once you no longer need it:
+```javascript
+
+/* start listening for messages */
+const hubListenerCancelToken = Hub.listen(/.*/, (data) => {
```
Comment (jimblanc) [inline src/fragments/lib-v1/utilities/js/hub.mdx:84]: Does Hub still return a cancellation token? Wasn't aware of this

Also can remove regex filtering

**Review by jimblanc — COMMENTED** [review]
Looks good, some comments on Hub

File: src/fragments/lib-v1/utilities/js/hub.mdx (hunk 7 above)
Comment (stocaaro) [reply src/fragments/lib-v1/utilities/js/hub.mdx:84]: Killing the regex here. It does return an cancellation function.

Comment (stocaaro) [reply src/fragments/lib-v1/utilities/js/hub.mdx:84]: In the new called its been renamed the `StopListenerCallback` by typing

**Review by jimblanc — APPROVED** [review]

**Pull request merged**